packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,119 @@
+import { Construct, Fn, Stack } from '@aws-cdk/core';
+import { AwsCustomResource, AwsCustomResourcePolicy, PhysicalResourceId } from '@aws-cdk/custom-resources';
+import { ILambdaDeploymentConfig } from './deployment-config';
+
+/**
+ * Lambda Deployment config type
+ */
+export enum CustomLambdaDeploymentConfigType {
+  /**
+   * Canary deployment type
+   */
+  CANARY = 'Canary',
+
+  /**
+   * Linear deployment type
+   */
+  LINEAR = 'Linear'
+}
+
+/**
+ * Properties of a reference to a CodeDeploy Lambda Deployment Configuration.
+ */
+export interface CustomLambdaDeploymentConfigProps {
+
+  /**
+   * The type of deployment config, either CANARY or LINEAR
+   */
+  readonly type: CustomLambdaDeploymentConfigType;
+
+  /**
+   * The integer percentage of traffic to shift:
+   * - For LINEAR, the percentage to shift every interval
+   * - For CANARY, the percentage to shift until the interval passes, before the full deployment
+   */
+  readonly percentage: string;
+
+  /**
+   * The interval, in number of minutes:
+   * - For LINEAR, how frequently additional traffic is shifted
+   * - For CANARY, how long to shift traffic before the full deployment
+   */
+  readonly interval: string;
+}
+
+/**
+ * A custom Deployment Configuration for a Lambda Deployment Group.
+ *
+ */
+export class CustomLambdaDeploymentConfig extends Construct implements ILambdaDeploymentConfig {
+
+  /**
+   * The name of the deployment config
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props: CustomLambdaDeploymentConfigProps) {
+    super(scope, id);
+    const deploymentType = 'TimeBased' + props.type.toString();
+    let routingConfig;
+    let customDeploymentConfigName;
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: props.interval,
+          canaryPercentage: props.percentage,
+        },
+      };
+      customDeploymentConfigName = 'LambdaCanary' + props.percentage.toString() + 'Percent' + props.interval.toString() + 'Minutes';
```

Comment:
What's happening here? CDK constructs compose, you don't need any of this craziness...

Reply:
I did it this way so that, in the event that the user changes the percentage/interval, the deployment config will be recreated with the new settings -- you can't modify existing configs.

Reply:
This is no longer used in any construct naming.